tests/performance/test_data_analysis/unit_tests/test_statistical_comparator.py

```diff
@@ -0,0 +1,81 @@
+# Copyright (C) 2015, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+"""Unit tests for the StatisticalComparator class"""
+
+import pytest
+import pandas as pd
+
+from statistical_data_analyzer import StatisticalComparator
+
+
+@pytest.fixture
+def sample_data1() -> pd.DataFrame:
+    """Fixture that returns a DataFrame for the first data set.
+
+    Returns:
+        (pd.Dataframe): Dataframe with the first data.
+    """
+    data = {
+        'Metric1': [10, 20, 30, 40],
+        'Metric2': [100, 200, 300, 400]
+    }
+    return pd.DataFrame(data)
+
+
+@pytest.fixture
+def sample_data2() -> pd.DataFrame:
+    """Fixture that returns a DataFrame for the second data set.
+
+    Returns:
+        (pd.Dataframe) Dataframe with the second data.
+    """
+    data = {
+        'Metric1': [120, 20, 30, 40],
+        'Metric2': [100, 200, 300, 400]
+    }
+    return pd.DataFrame(data)
+
+
+def test_calculate_basic_statistics(sample_data1: pd.DataFrame) -> None:
+    """Test that checks the accuracy of the statistics calculation.
+
+    Args:
+        sample_data1 (pd.Dataframe): first Dataframe.
+    """
+    comparator = StatisticalComparator()
+    mean_value = comparator.calculate_basic_statistics(sample_data1, 'Metric1', 'Mean')
+    median_value = comparator.calculate_basic_statistics(sample_data1, 'Metric1', 'Median')
+    max_value = comparator.calculate_basic_statistics(sample_data1, 'Metric2', 'Max value')
+    min_value = comparator.calculate_basic_statistics(sample_data1, 'Metric2', 'Min value')
+    std_value = comparator.calculate_basic_statistics(sample_data1, 'Metric1', 'Standard deviation')
+    var_value = comparator.calculate_basic_statistics(sample_data1, 'Metric2', 'Variance')
+    assert mean_value == 25
+    assert median_value == 25
+    assert max_value == 400
+    assert min_value == 100
+    assert std_value == 12.91
+    assert var_value == 16666.67
+
+
+def test_comparison_basic_statistics(sample_data1: pd.DataFrame, sample_data2: pd.DataFrame) -> None:
+    """Test that checks the accuracy of the statistics comparison.
+
+    Args:
+        sample_data1 (pd.Dataframe): first Dataframe.
```

Comment:
Missing sample_data2 arguments in the docstring

Reply:
Solved in https://github.com/wazuh/wazuh-qa/pull/5622/commits/a8be0593badbbd04bb3211468f64ad2689d17bf3